src/Hazelcast.Net/CP/IFencedLock.cs

```diff
@@ -19,104 +19,126 @@
+        /// Acquires the lock only if it is free or already held by the current context at the time of invocation
+        /// and the acquire limit is not exceeded, and returns the fencing token assigned to the current context
         /// for this lock acquire. If the lock is acquired reentrantly, the same fencing token is returned. 
         /// If the lock is already held by another caller or the lock acquire limit is exceeded, then this method 
         /// immediately returns <see cref="InvalidFence"/> that represents a failed lock attempt.
         /// </summary>
         /// <param name="timeout">The maximum time to wait for the lock</param>
+        /// <param name="lockContext">The <see cref="LockContext"/>.</param>
         /// <returns>The fencing token if the lock was acquired and <see cref="InvalidFence"/> otherwise</returns>
-        Task<long> TryLockAndGetFenceAsync(TimeSpan timeout);
+        Task<long> TryLockAndGetFenceAsync(LockContext lockContext, TimeSpan timeout);
 
         /// <summary>
-        /// Acquires the lock only if it is free or already held by the current <see cref="AsyncContext"/> at the time of invocation
-        /// and the acquire limit is not exceeded, and returns the fencing token assigned to the current <see cref="AsyncContext"/>
+        /// Acquires the lock only if it is free or already held by the current context at the time of invocation
+        /// and the acquire limit is not exceeded, and returns the fencing token assigned to the current context
         /// for this lock acquire. If the lock is acquired reentrantly, the same fencing token is returned. 
         /// If the lock is already held by another caller or the lock acquire limit is exceeded, then this method 
         /// immediately returns <see cref="InvalidFence"/> that represents a failed lock attempt.
         /// </summary>
+        /// <param name="lockContext">The <see cref="LockContext"/>.</param>
         /// <returns>The fencing token if the lock was acquired and <see cref="InvalidFence"/> otherwise</returns>
-        Task<long> TryLockAndGetFenceAsync();
+        Task<long> TryLockAndGetFenceAsync(LockContext lockContext);
 
         /// <summary>
-        /// Releases the lock if the lock is currently held by the current <see cref="AsyncContext"/>.
+        /// Releases the lock if the lock is currently held by the current context.
         /// </summary>
-        Task UnlockAsync();
+        /// <param name="lockContext">The <see cref="LockContext"/>.</param>
+        Task UnlockAsync(LockContext lockContext);
 
         /// <summary>
         /// Checks whether this lock is locked or not.
         /// </summary>
-        /// <returns><see cref="true"/> if the lock was acquired otherwise <see cref="false"/></returns>
+        /// <returns><c>true</c> if the lock was acquired; otherwise <c>false</c>.</returns>
         Task<bool> IsLockedAsync();
```

Comment:
```suggestion
        Task<bool> IsLockedAsync(LockContext lockContext);
```

Reply:
won't change see other comment re. threading and java